### PR TITLE
Features/docu add math description caes

### PR DESCRIPTION
**Pull request by jakob-wo**
* This PR adds a mathematical description to the docstring of `custom.GenericCAESBlock`
  * It Includes a list of created variables, of created constraints and a table

* Related issues: This PR fulfils a task of #428 and follows the docstring template recommedation made in #544

* It bases on the paper:
    Kaldemeyer, C.; Boysen, C.; Tuschy, I.
    A Generic Formulation of Compressed Air Energy Storage as
    Mixed Integer Linear Program – Unit Commitment of Specific
    Technical Concepts in Arbitrary Market Environments
    Materials Today: Proceedings 00 (2018)

### CODE REVIEW

File: oemof/solph/custom.py [hunk 1]
```diff
@@ -412,7 +412,225 @@ def constraint_group(self):
 
 
 class GenericCAESBlock(SimpleBlock):
-    """Block for nodes of class:`.GenericCAES`."""
+    r"""Block for nodes of class:`.GenericCAES`.
+
+    **The following variables are created:**
```
Comment (p-snft) [inline oemof/solph/custom.py]: This is already part of the table. I think, it is not needed here.

Comment (p-snft) [inline oemof/solph/custom.py]: This is already part of the table. I think, it is not needed here.

**Comment by jakob-wo** [comment]
Thanks for your feedback!

I will try to reply to every aspect you mentioned:

1) List of Variables: I agree on @p-snft comment. The list is a partial repetition of the table below and therfore we may leave it out in the docstring. But in that case I suggest that the Table, containing variables (V) and parameters (P), shall always be ordered according to the the "type", i.e. V or P. So one can get a quick overview of the created variables. Here, in case of teh CAES this is already done and I will remove the List of variables.

2) Moving the code to components: In my opinion we shout leave it in custom. For two reasons: a) The component and its code as it is now is not well enough documented and an example is missing.  I could go on improving it but would need help from @ckaldemeyer as a few aspects are still not clear to me. b) I see solph.custon as a place not only for "not-100%-tested-and-documented-Components" but also for highly specified Components that have no generic formulation or application. And I see the CAES as a highly specified component that fits very well in custom. Parts of this topic are related to #588 and we shall continue the discussion there, I suggest.

3) Size of variables in the table: Yes, that is due to the size of the indices. I tried to pre-define the column-width but it did not work (I did not spend much time on this). If anyone knows a solution or work-around... let me know, please.

4) "intersection" instead of "intersept": Yes, you are right! I will change that. That was copy/paste from the paper, so @ckaldemeyer you might want to correct that typo in your paper as well ;-)

5) paper is necessary to use the component: Yes, I agree on that, too. For me it was just adding the constraints and variables and I would prefere to finish and merge that part first.

**Comment by jakob-wo** [comment]
> 
> There are so many off-set linear equations. Is it possible to have a short summary alread in the table at the parameters of all the linear equations?
>

I did not find a nice and consistent way to avoid naming all paramters in the table. I left it as it was before but I am open to suggestions.

**Pull request merged**